scvi/hub/hub_metadata.py

```diff
@@ -0,0 +1,188 @@
+import json
+from dataclasses import dataclass
+from typing import List, Optional
+
+import torch
+from dataclasses_json import dataclass_json
+from huggingface_hub import ModelCard, ModelCardData
+
+from ._constants import (
+    ANNDATA_VERSION_TAG,
+    ANNOTATED_TAG,
+    DEFAULT_MISSING_FIELD,
+    DEFAULT_NA_FIELD,
+    DEFAULT_PARENT_MODULE,
+    HF_LIBRARY_NAME,
+    MODALITY_TAG,
+    MODEL_CLS_NAME_TAG,
+    SCVI_VERSION_TAG,
+    TISSUE_TAG,
+)
+from ._utils import _get_counts_and_latent_info
+from .model_card_template import template
+
+
+@dataclass_json
+@dataclass
+class HubMetadata:
+    """Placeholder docstring. TODO complete."""
+
+    data_cell_count: int
+    data_gene_count: int
+    scvi_version: str
+    anndata_version: str
+    large_data_url: Optional[str] = None
+    model_parent_module: str = DEFAULT_PARENT_MODULE
+
+    @classmethod
+    def from_dir(
+        cls,
+        local_dir: str,
+        anndata_version: str,
+        data_cell_count: Optional[int] = None,
+        data_gene_count: Optional[int] = None,
+        **kwargs,
+    ):
+        """Placeholder docstring. TODO complete."""
+        cell_count, gene_count = _get_counts_and_latent_info(
+            local_dir, data_cell_count, data_gene_count
+        )
+        torch_model = torch.load(f"{local_dir}/model.pt")
+        scvi_version = torch_model["attr_dict"]["registry_"]["scvi_version"]
+
+        return cls(
+            cell_count,
+            gene_count,
+            scvi_version,
+            anndata_version,
+            **kwargs,
+        )
+
+
+class HubModelCardHelper:
```

Comment:
I ended up decoupling the model card completely from the required metadata. The latter might be a subset of the former or grow to be a superset or it might just have no relationship to it. Conceptually these are two different things. One -- the model card -- is the human-readable human0facing description of the model. The other -- the metadata -- is some required bits of info needed for scvi-hub to work. There are essentially three options:
1. Embed the metadata in the model card, and ask users not to edit those parts. Essentially this constraints the human-facing contents of the model card. And for that reason, I dont love it.
2. Embed the metadata either in the header or in a commented section at the bottom of the model card. This could work but I prefer 3.
3. What I did, aka decouple the model card completely from the metadata. This makes sense conceptually, is the least error prone in terms of accidental alteration, and avoid having to parse text. Lastly, if the user wants, they can have no model card at all. That's allowed on huggingface.
Happy to chat more today.